narwhals/_duckdb/expr.py

```diff
@@ -438,6 +471,40 @@ def null_count(self: Self) -> Self:
             "null_count",
         )
 
+    def over(
+        self: Self,
+        partition_by: Sequence[str],
+        order_by: Sequence[str] | None,
+    ) -> Self:
+        if self._backend_version < (1, 3):
+            msg = "At least version 1.3 of DuckDB is required for `over` operation."
+            raise NotImplementedError(msg)
+        if (window_function := self._window_function) is not None:
```

Comment:
AFAIK, you only need the `is not None` when the type overrides `__bool__`. 

Going to guess this is from experience battling `pandas`, `polars`, and `numpy` who all tell you off for using `__bool__`

```suggestion
        if (window_function := self._window_function):
```